### PR TITLE
On hold feature

### PR DESCRIPTION
**Pull request by deniss-mc**
**Jira issue:**
https://glia.atlassian.net/browse/MUIC-302

**Additional info:**
contains:
https://glia.atlassian.net/browse/MUIC-648
https://glia.atlassian.net/browse/MUIC-714

**Screenshots:**

### CODE REVIEW

**Review by gugalo — APPROVED** [review]
Some minor comments

File: widgetssdk/src/main/java/com/glia/widgets/call/CallState.java [hunk 1]
```diff
@@ -307,47 +316,51 @@ public CallState speakerValueChanged(boolean isSpeakerOn) {
...
-                isMuted == callState.isMuted &&
-                hasVideo == callState.hasVideo &&
-                Objects.equals(callStatus, callState.callStatus) &&
-                Objects.equals(queueTicketId, callState.queueTicketId) &&
-                Objects.equals(companyName, callState.companyName) &&
-                Objects.equals(requestedMediaType, callState.requestedMediaType) &&
-                isSpeakerOn == callState.isSpeakerOn;
+    public boolean isMuteButtonEnabled() {
+        return (isAudioCall() || is2WayVideoCall()) && !isOnHold;
     }
```
[the remark this answers is not in the document]
Comment (deniss-mc) [reply widgetssdk/src/main/java/com/glia/widgets/call/CallState.java:328]: It was requirement - we will modify it further to indicate that user is
-> audio.paused(muted)
-> video.paused(will require us to put video on pause manually on hold event)

File: widgetssdk/src/main/java/com/glia/widgets/view/floatingvisitorvideoview/FloatingVisitorVideoConstraintLayout.java [hunk 2]
```diff
@@ -1,40 +1,35 @@
...
 import androidx.core.util.Pair;
 
 import com.glia.androidsdk.comms.VisitorMediaState;
 import com.glia.widgets.R;
 import com.glia.widgets.di.Dependencies;
+import com.glia.widgets.helper.Utils;
 import com.glia.widgets.view.ViewHelpers;
 
-public class FloatingVisitorVideoCoordinatorLayout extends FrameLayout implements FloatingVisitorVideoContract.View {
+public class FloatingVisitorVideoConstraintLayout extends ConstraintLayout implements FloatingVisitorVideoContract.View {
```
Comment (jakob-mc) [inline widgetssdk/src/main/java/com/glia/widgets/view/floatingvisitorvideoview/FloatingVisitorVideoConstraintLayout.java:19]: NIT: If you omit the view internal implementation (Coordinator, Constraint etc) from the name, then you don't have so many renaming changes next time the implementation is changed. Are there any benefits? 

Why not just `FloatingVisitorVideoView`?

Comment (jakob-mc) [reply widgetssdk/src/main/java/com/glia/widgets/view/floatingvisitorvideoview/FloatingVisitorVideoConstraintLayout.java:19]: Ok, I see you already have a view with that name, but `FloatingVisitorVideoContainer` or smthng?

File: widgetssdk/src/main/java/com/glia/widgets/view/floatingvisitorvideoview/FloatingVisitorVideoView.java [hunk 3]
```diff
@@ -48,19 +44,31 @@ public void onDestroy() {
...
+        resumeVideoStream();
+        onHoldOverlay.setVisibility(GONE);
+    }
+
+    public boolean hasVideo() {
+        return videoView != null;
     }
 
     private void init() {
         LayoutInflater.from(getContext()).inflate(R.layout.visitor_video_floating_view, this);
```
Comment (jakob-mc) [inline widgetssdk/src/main/java/com/glia/widgets/view/floatingvisitorvideoview/FloatingVisitorVideoView.java:70]: I don't see a system in view and it's layout xml naming. Usually it's a good practice in xml name to have `activity_`, `view_` prefix. Is there any convention?

**Review by jakob-mc — CHANGES_REQUESTED** [review]

File: widgetssdk/src/main/java/com/glia/widgets/view/floatingvisitorvideoview/FloatingVisitorVideoView.java (hunk 3 above)
Comment (deniss-mc) [reply widgetssdk/src/main/java/com/glia/widgets/view/floatingvisitorvideoview/FloatingVisitorVideoView.java:70]: If you use it as postfix you can have grouped views. I am following order that was there initially.

Could be nice to call parent group layouts layouts (containers) and well views remain views, activities activities etc.

But you can suggest changes :)

File: widgetssdk/src/main/java/com/glia/widgets/view/floatingvisitorvideoview/FloatingVisitorVideoConstraintLayout.java (hunk 2 above)
Comment (deniss-mc) [reply widgetssdk/src/main/java/com/glia/widgets/view/floatingvisitorvideoview/FloatingVisitorVideoConstraintLayout.java]: renamed

File: widgetssdk/src/main/java/com/glia/widgets/call/CallView.java [hunk 4]
```diff
@@ -799,7 +785,7 @@ private void hideUIOnCallEnd() {
         setVisibility(INVISIBLE);
         Activity activity = Utils.getActivity(this.getContext());
         hideOperatorVideo();
-        hideVisitorVideo();
+        floatingVisitorVideoCoordinatorLayout.hide();
```
[the remark this answers is not in the document]
Comment (deniss-mc) [reply widgetssdk/src/main/java/com/glia/widgets/call/CallView.java]: extracted

File: widgetssdk/src/main/java/com/glia/widgets/view/floatingvisitorvideoview/FloatingVisitorVideoView.java (hunk 3 above)
Comment (jakob-mc) [reply widgetssdk/src/main/java/com/glia/widgets/view/floatingvisitorvideoview/FloatingVisitorVideoView.java:70]: ok let's keep it postfixed then for now, as this seems to be the convention here.

but there is still inconsistency in the class vs layout names: `FloatingVisitorVideo...` vs `visitor_video_floating..`

**Review by jakob-mc — APPROVED** [review]

**Pull request merged**